vnctl/lib/vnctl/webapi.rb

```diff
@@ -2,7 +2,44 @@
 
 module Vnctl
   class WebApi
-    include HTTParty
-    base_uri "#{Vnctl.conf.webapi_uri}:#{Vnctl.conf.webapi_port}"
+    def initialize
+      @webapi_uri = Vnctl.conf.webapi_uri
+      @webapi_port = Vnctl.conf.webapi_port.to_i
+      @webapi_version = Vnctl.conf.webapi_version
+      @webapi_protocol = Vnctl.conf.webapi_protocol
+      @output_format = Vnctl.conf.output_format
+
+      @webapi_full_uri = "%s://%s:%s/api/%s" %
+        [@webapi_protocol, @webapi_uri, @webapi_port, @webapi_version]
+    end
+
+    def send_request(verb, suffix, params = nil)
+      uri = URI("#{@webapi_full_uri}/#{suffix}.#{@output_format}")
+
+      uri.query = URI.encode_www_form(params) if params
+
+      response = Net::HTTP.start(uri.host, uri.port) do |http|
+        request = verb.new(uri.request_uri)
+        http.request(request)
+      end
+
+      response.body
+    end
+
+    def post(suffix, params = nil)
```

Comment:
This reek is also fine. There is one main method and then 4 shortcuts that fill in the `verb` parameter.